subsys/bluetooth/mesh/blob_srv.c

```diff
@@ -844,6 +844,19 @@ static int blob_srv_settings_set(struct bt_mesh_model *mod, const char *name,
 		return 0;
 	}
 
+	/* If device restarted before it handled `XFER_START` server doesn't have transfer
+	 * to be restored - transfer should be abandoned and new one must be started
+	 */
```

Comment:
Perhaps we should not store it here? https://github.com/zephyrproject-rtos/zephyr/blob/9ca9b91b1330d982ef24f610992682951ca85934/subsys/bluetooth/mesh/blob_srv.c#L914-L933

Or restore BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START phase?

Reply:
I took this option, I restore `BT_MESH_BLOB_XFER_PHASE_WAITING_FOR_START` phase in `blob_srv_settings_set`. This allows to continue as normal, no need to call `bt_mesh_blob_srv_recv` again, server is able to accept start normally.